src/persist/src/indexed/mod.rs

```diff
@@ -386,22 +348,17 @@ impl<U: Buffer, L: Blob> Indexed<U, L> {
             // future's seqno_upper and that there is nothing for that future in
             // [future.seqno_upper, self.futures_seqno_upper). Use this to make the
             // seqnos of all the future batches line up.
-
-            // We cannot assert strict equality here because a buffer may have
-            // previously failed to truncate.
-            debug_assert!(desc.start <= self.futures_seqno_upper);
             let new_start = future.seqno_upper()[0];
             debug_assert!(new_start <= self.futures_seqno_upper);
```

Comment:
I reworded a lot of this comment because its a lot simpler without buffer (which could be lagging behind on truncation so the previous commit had to sometimes adjust the start to be greater than it was)

in the old world with fallible buffer truncation you could have:

1. read from seqno [0, 3) in buffer and place it in future
2. truncate fails. the index's future seqno upper (aka global max ingested seqno frontier is 3, as is the futures seqno upper)
3. read from buffer again. buffer gives you [0, 6) because it lacks the api to filter for you atm
4. we only want to read from [3, 6) because we already ingested [0, 3). 

The key thing the invariant was checking in the previous pr is -- let's make sure we don't bump the sequence number past the index's future seqno_upper.

Previously, before we ran step in the background, buffer truncate would never fail, so the desc.start always equalled index.futures_seqno_upper and we would sometimes push desc.start back to make the batches line up. With fallible truncate the "push desc.start back, or not at all" property goes away.

This is all quite a bit simpler without buffer where we only really have to check "is the allocated sequence number greater than any that came before?"